lib/internal/test_runner/reporter/tap.js

```diff
@@ -224,8 +224,8 @@ function jsToYaml(indent, name, value, seen) {
     }
 
     if (errIsAssertion) {
-      result += jsToYaml(indent, 'expected', errExpected, seen);
-      result += jsToYaml(indent, 'actual', errActual, seen);
+      result += jsToYaml(indent, 'expected', errExpected, new SafeSet(seen));
+      result += jsToYaml(indent, 'actual', errActual, new SafeSet(seen));
```

Comment:
Can we re-use the same `SafeSet`? If not, can we add a comment?

Reply:
@adu95 comment added now in 352a55685c3562e20eb1bc12048b400c087dc396.